pkg/controller/logstorage/linseed/eskubecontrollers.go

```diff
@@ -29,37 +30,54 @@ import (
 	"github.com/tigera/operator/pkg/render"
 	"github.com/tigera/operator/pkg/render/kubecontrollers"
 	"github.com/tigera/operator/pkg/tls/certificatemanagement"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"k8s.io/apimachinery/pkg/api/errors"
 )
 
-func (r *ReconcileLogStorage) createESKubeControllers(
+// TODO: Move this out of this controller.
```

Comment:
What else does es-kube-controllers do?

- Provisions users to ES, and generates usernames / passwords for accessing ES via ESGW
- Copies a public certificate to managed clusters so they can verify ESGW as part of the TLS handshake.
- ???

I think that means if ESGW gets removed, we can get rid of ESKC as well! But looking for input in case anyone is aware of other es-kube-controllers responsibilities...

Reply:
> Copies a public certificate to managed clusters so they can verify ESGW as part of the TLS handshake.

The controller you created for linseed would replace this logic once we remove ESGW, right?

I think that kube controllers also passes a config map containing logstorage configuration needed in the managed cluster. It modifies (or adds, can't remember if it exists before it's modified) the cluster name so the managed cluster knows what "midfix" it should use for flow logs.

Look [here](https://github.com/tigera/calico-private/blob/master/kube-controllers/pkg/controllers/elasticsearchconfiguration/reconcile_elasticsearch_configuration.go#L90) and [here](https://github.com/tigera/calico-private/blob/master/kube-controllers/pkg/controllers/elasticsearchconfiguration/commonbuild.go).

Reply:
> It modifies (or adds, can't remember if it exists before it's modified) the cluster name so the managed cluster knows what "midfix" it should use for flow logs.

This is no longer needed with Linseed.